### PR TITLE
AF: vol-11x + eval-every-3 + WD=5e-3 (moderate vol + lighter regularization)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Combine moderate vol-weight (11x) with lighter WD (5e-3 vs champion 1e-2) and eval-every-3 throughput trick. Rei #3238 showed WD=5e-3 gives programme-best surface (0.000249) but vol regressed with 10x weight. Higher vol emphasis (11x) + WD=5e-3 may find the dual-metric sweet spot. Vol-13x/14x confirmed unstable; 11x is the safe zone.

## Instructions
```bash
cd target/icml2026 && python train.py --dataset airfrans --optimizer lion --lr 1.25e-4 \
  --cosine-t-max 50 --grad-clip 1.0 --weight-decay 5e-3 \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 --enable-fourier \
  --epochs 999 --ema-decay 0.999 --use-lookahead --compile-model \
  --vol-loss-weight 11 --eval-interval 3 --save-checkpoint \
  --wandb-group af-eval-every-3-vol-sweep
```
**Target:** surface_mse < 0.000296 AND vol_mse < 0.002039

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT gohan:

## Results

### Best Metrics (epoch 204)

| Metric | Result | Baseline | Target | Delta |
|--------|--------|----------|--------|-------|
| `val_primary/surface_mse` | **0.000776** | 0.000296 | < 0.000296 | +162% (MISSED) |
| `full_val/volume_mse` | **0.005042** | 0.002039 | < 0.002039 | +147% (MISSED) |

### Surface MSE Breakdown (ep204)
- Ux: 7.0e-06
- Uy: 1.0e-06
- nut: 1.0e-06
- p: 3.094e-03

### Volume MSE Breakdown (ep204)
- Ux: 1.044e-02
- Uy: 6.778e-03
- nut: 2.231e-03
- p: 7.24e-04

### Training Command
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --optimizer lion --lr 1.25e-4 \
  --cosine-t-max 50 --grad-clip 1.0 --weight-decay 5e-3 \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 --enable-fourier \
  --epochs 999 --ema-decay 0.999 --use-lookahead --compile-model \
  --volume-loss-weight 11 --eval-interval 3 --save-checkpoint \
  --wandb-group af-eval-every-3-vol-sweep \
  --agent gohan --wandb-name "gohan/af-vol11x-eval3-wd5e3"
```

### W&B Run
- Run ID: `5uyihi7e`
- Link: https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/5uyihi7e

### Peak Memory
- ~54 GB GPU VRAM (GPU 0, RTX PRO 6000 Blackwell 98 GB)

### What happened

**The experiment diverged catastrophically at epoch ~220 and did not beat either target.**

The model converged well through the first 4 cosine cycles (T_max=50), with steady improvement:
- ep51: surface=0.00776, vol=0.0286
- ep120: surface=0.00187, vol=0.00780
- ep204: surface=0.000776, vol=0.00504 (best checkpoint)

At epoch 220, gradient norms spiked from ~100 to 520, then to 879, then escalated to 1M+ by epoch 235. Loss went from 0.18 to 25+. The model never recovered.

**Root cause analysis:**
1. **Lion optimizer instability with 4L depth**: The baseline champion uses 2L/256d with AdamW. The 4L depth creates longer gradient paths that amplify instability at cosine LR peaks. Lion's sign-based updates are more sensitive to gradient explosions than AdamW's adaptive scaling.
2. **WD=5e-3 insufficient regularization**: The champion uses WD=1e-2. Lighter regularization allows weights to grow larger, making the model more vulnerable to gradient explosions during cosine peaks.
3. **grad-clip=1.0 insufficient**: With 4L depth and Lion, gradient norms regularly exceeded 100 even in healthy training. A clip of 1.0 was not aggressive enough to prevent the cascade once instability began.

**Note on flag mismatch**: The PR instructions used `--vol-loss-weight 11` but the actual CLI flag is `--volume-loss-weight`. Corrected in the actual run.

### Suggested follow-ups
- **Try 2L/256d with vol-weight=11 + WD=5e-3**: The 4L depth seems to be the instability factor. The baseline champion's 2L architecture may be more stable with this vol/WD combination.
- **Try lower grad-clip (0.5)**: gc=0.5 has historically stabilized AirfRANS runs. This might prevent the gradient cascade.
- **Try WD=7.5e-3 as middle ground**: Between the champion's 1e-2 and this experiment's 5e-3 — lighter regularization without the instability risk.
- **Try Lion with 2L**: The optimizer change (AdamW→Lion) combined with depth change (2L→4L) makes it hard to attribute failure. Isolating Lion at 2L would clarify if the optimizer or depth is the issue.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3356 because ADVISOR: Closing — WRONG CONFIG: used Lion 4L/256d instead of AdamW 2L/256d AF champion. Same systematic error as #3338/#3343/#3355. Diverged ep220 with gradient explosion. Reassigning with correct config..
